app/src/main/java/net/bible/android/view/activity/readingplan/DailyReading.kt

```diff
@@ -313,29 +464,56 @@ class DailyReading : CustomTitlebarActivityBase(R.menu.reading_plan) {
             R.id.done -> {
                 Log.i(TAG, "Force Done")
                 try {
-                    readingPlanControl.done(mReadings.readingPlanInfo, mDay, true)
-                    updateTicksAndDone()
+                    done(true)
+                    updateTicksAndDone(readingPlanOneDay)
                 } catch (e: Exception) {
                     Log.e(TAG, "Error when Done daily reading", e)
                     Dialogs.getInstance().showErrorMsg(R.string.error_occurred, e)
                 }
 
             }
             R.id.reset -> {
-                readingPlanControl.reset(mReadings.readingPlanInfo)
-                finish()
+                Dialogs.getInstance().showMsg(R.string.reset_plan_question,true)
+                    {
+                        Log.d(TAG, "Resetting plan id $readingPlanMetaID")
+                        dbAdapter.resetPlan(readingPlanMetaID!!)
+                        finish()
+                    }
+
                 isHandled = true
             }
             R.id.setStartToJan1 -> {
-                readingPlanControl.setStartToJan1(mReadings.readingPlanInfo)
+                dbAdapter.setPlanStartDate(readingPlanMetaID!!, DateUtils.truncate(Date(), Calendar.YEAR))
 
-                // refetch readings for chosen day
-                mReadings = readingPlanControl.getDaysReading(mDay)
+                loadPlanOneDay(null, ReadingPlanInformationDB(readingPlanMetaID), true)
 
-                // update date and day no
-                date.text = mReadings.readingDateString
-                day.text = mReadings.dayDesc
+                isHandled = true
+            }
+            R.id.importNewPlan -> {
```

Comment:
Manual importing new plans should be done via ACTION_GET_CONTENT intent (similarly as in InstallZip). Then this does not need any permissions at all and user can choose importable file from any source (like google drive).

Reply:
Done